services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/azure/BlobFileClientITest.kt

```diff
@@ -41,7 +41,7 @@ class BlobFileClientITest : FileClientTest() {
         azuriteContainer.start()
         val connectionString =
             "BlobEndpoint=http://${azuriteContainer.host}:${azuriteContainer.firstMappedPort}/devstoreaccount1;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;"
-        serviceClient = AzureContainerServiceClient(connectionString).blobServiceClient()
+        serviceClient = AzureContainerServiceClient(connectionString, "").blobServiceClient()
```

Comment:
Could the second parameter (`endpoint`) have a default value or could there be a second constructor in order to use it here like `AzureContainerServiceClient(connectionString)`?

Reply:
Adding a default value causes the application startup to fail because Spring can't find a default constructor. I think the default value gets compiled into several constructors.

To make this neater, there would have to be a default (empty) constructor in AzureContainerServiceClient, which would mean it would have to use field injection instead of constructor injection. I don't see this as an overall improvement.

Reply:
Sounds reasonable